staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -978,16 +975,19 @@ func getNodePredicatesForService(service *v1.Service) []NodeConditionPredicate {
 	return allNodePredicates
 }
 
+// This predicate just validates if the providerID has been set and triggers a
+// node sync. It is _not_ used when determining which nodes to use when
+// configuring the load balancer's backend pool.
+func nodeHasProviderIDPredicate(node *v1.Node) bool {
```

Comment:
We only use this in `shouldSyncUpdatedNode()` now, and we test whether the predicate changed.  IOW we handle "" to "value" and "value" to "", but not "foo" to "bar".  Should we just compare that ?

Reply:
Can the providerID even change? I was under the impression it can't